pyfr/readers/base.py

```diff
@@ -150,13 +150,19 @@ def _pair_periodic_fluid_faces(self, bpart, resid):
                 rfidx = fuzzysort(rfpts.mean(axis=1).T, range(len(rfnodes)))
 
                 for lfn, rfn in zip(lfnodes[lfidx], rfnodes[rfidx]):
-                    lf = resid.pop(tuple(sorted(lfn)))
-                    rf = resid.pop(tuple(sorted(rfn)))
+                    lf = list(resid.pop(tuple(sorted(lfn))))
```

Comment:
`resid.pop(tuple(sorted(lfn)))[:-1] + (int(k) + 1,)` allows you to work with tuples directly.